### PR TITLE
Adapt sql.js driver for new persistence layer

### PR DESCRIPTION
**Pull request by daniel-lang**
This is the sql.js driver with small adjustments for the new persistence layer. I just merged the master branch instead of just the sql.js branch. I hope, that's okay.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -226,6 +227,10 @@ export function getMongoManager(connectionName: string = "default"): MongoEntity
     return getConnectionManager().get(connectionName).manager as MongoEntityManager;
 }
 
+export function getSqljsManager(connectionName: string = "default"): SqljsEntityManager {
```
Comment (pleerock) [inline src/index.ts]: don' t forget about comments 😉

File: src/driver/sqljs/SqljsDriver.ts [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+    /**
+     * Creates generated map of values generated or returned by database after INSERT query.
+     */
+    createGeneratedMap(metadata: EntityMetadata, insertResult: any) {
+        const generatedMap = metadata.generatedColumns.reduce((map, generatedColumn) => {
+            // seems to be the only way to get the inserted id, see https://github.com/kripken/sql.js/issues/77
+            if (generatedColumn.isPrimary && generatedColumn.generationStrategy === "increment") {
+                const query = "SELECT last_insert_rowid()";
+                let result = this.databaseConnection.exec(query);
```
Comment (pleerock) [inline src/driver/sqljs/SqljsDriver.ts]: Query Execution is synchronous?...

Comment (daniel-lang) [reply src/driver/sqljs/SqljsDriver.ts]: Sadly yes. I have to fake the asynchronous behavior with this driver

**Comment by daniel-lang** [comment]
@pleerock your comments are adressed

**Comment by pleerock** [comment]
Thank you for your hard work!

**Pull request merged**